Tests/ValetIntegrationTests/ValetIntegrationTests.swift

```diff
@@ -481,6 +482,57 @@ class ValetIntegrationTests: XCTestCase
             }
         }
     }
+
+    // MARK: codable(forKey:)
+
+    func test_codableForKey_succeedsForValidKey() throws {
+        try allPermutations.forEach { valet in
+            try valet.setCodable(stringArray, forKey: key)
+            XCTAssertEqual(stringArray, try valet.codable(forKey: key))
+        }
+    }
+
+    func test_codableForKey_throwsDecoderErrorOnInvalidType() throws {
+        try allPermutations.forEach { valet in
+            try valet.setCodable(stringArray, forKey: key)
+            XCTAssertThrowsError(try valet.codable(forKey: key) as [Int]) { error in
+                switch error {
+                case DecodingError.typeMismatch:
+                    break // expected
+
+                default:
+                    XCTFail("Unexpected error: \(error)")
+                }
```

Comment:
I think the following will compile and will be a little more concise without losing readability:
```suggestion
                XCTAssertEquals(error as? DecodingError, .typeMismatch)
```

Reply:
This suggestion did not work, I'm getting 2 errors:
1. `Cannot convert value of type '(Any.Type, DecodingError.Context) -> DecodingError' to expected argument type 'DecodingError?'`
2. `Global function 'XCTAssertEqual(_:_:_:file:line:)' requires that 'DecodingError' conform to 'Equatable'`